test/aoc.test.js

```diff
@@ -202,6 +205,52 @@ let res = func({input, udf})
 expect(res).toBe(4361)
 })
 
+test("day4-part1", () => {
+  let input = `Card 1: 41 48 83 86 17 | 83 86  6 31 17  9 48 53
+Card 2: 13 32 20 16 61 | 61 30 68 82 17 32 24 19
+Card 3:  1 21 53 59 44 | 69 82 63 72 16 21 14  1
+Card 4: 41 92 73 84 69 | 59 84 76 51 58  5 54 83
+Card 5: 87 83 26 28 32 | 88 30 70 12 93 22 82 36
+Card 6: 31 18 13 56 72 | 74 77 10 23 35 67 36 11`
+
+  let udf = {
+    // udf.toNum1 deals with empty strings after splitting by " ",
+    // otherwise toNum converts "" to 0
+    toNum1: x => x === "" ? undefined : udf.toNum(x),
+    // We want to ignore the undefined key in the freq object
+    removeUndef: o => {
+      delete o[undefined]
+      return o
+    },
+    ...udf_stdlib
+  }
+  
+  let line = rh`.input | udf.split "\\n" | .*line
+                       | udf.split ":" | .1`
+ 
+  // toNum1 could result in undefined values
+  let number = rh`${line} | udf.split " " | .*nums
+                          | udf.toNum1`
+
```

Comment:
1. Do we need to split by "|" at some point?
2. I think it would be preferable if we could split the input so that we are left precisely with the numbers, rather than having to eliminate empty strings/undefined later. Could we do this by splitting with ": ", " | ", and Regex(" +") ? Or perhaps using .trim()?

Reply:
Re 2: or perhaps parse the numbers using .match(/(\d+)/g) rather than split

Reply:
Using trim() would still leave some empty strings in the middle for lines like this:
```
41 92 73 84 69 | 59 84 76 51 58  5 54 83
```
There will be an empty string between 58 and 5 after splitting by spaces. `'58', '', '5'`

I think using match and regex would be a good idea.

Reply:
Yes, makes sense

Reply:
If we use .match, do we still need to split by "|"? It should be unecessary but the input is not splitted properly.

Reply:
It seems cleaner to have the "|" split, even though not strictly necessary. Let's also add a comment how the algorithm works -- the underlying assumption seems to be that each half can only mention a given number once, so instead of doing a join we can just look for num occurrences = 2